src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -202,24 +202,26 @@ internal static void CheckForSevereException(Exception e)
 
             System.Threading.Thread.CurrentThread.Name = "ConsoleHost main thread";
 
-            s_theConsoleHost = ConsoleHost.CreateSingletonInstance(configuration);
-            s_theConsoleHost.BindBreakHandler();
-
-            PSHost.IsStdOutputRedirected = Console.IsOutputRedirected;
-
-            if (args == null)
+            // We might be able to ignore console host creation error if we are running in 
+            // server mode, which does not require a console.
+            HostException hostException = null;
+            try
             {
-                args = new string[0];
+                s_theConsoleHost = ConsoleHost.CreateSingletonInstance(configuration);
```

Comment:
No, because otherwise CommandLineParameterParser will never be called and we can't find out if we are running server mode where s_theConsoleHost is not needed.  Also I am saving the HostException so I can re-throw if it turns out we are not running in server mode.

---
In reply to: [85440668](https://github.com/PowerShell/PowerShell/pull/2538#discussion_r85440668) [](ancestors = 85440668)